### PR TITLE
feat: use solhint features

### PR DESCRIPTION
**Pull request by 0xShaito**
New features in solhint allows us to implement different rules for different folders. Now we have a general one and a specific one on each folder. With this, we remove a lot of defaults that are not needed anymore and create a separation of rules

### CODE REVIEW

**Review by ashitakah — DISMISSED** [review]
🔥

File: src/interfaces/.solhint.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "rules": {
+    "ordering": "off",
```
Comment (0xShaito) [inline src/interfaces/.solhint.json:3]: Should we keep ordering on in interfaces?

File: .solhint.json [hunk 2]
```diff
@@ -1,18 +1,13 @@
 {
   "extends": "solhint:recommended",
   "rules": {
```
Comment (0xJabberwock) [inline .solhint.json:3]: I vote for warning on [`named-parameters-mapping`](https://github.com/solhint-community/solhint-community/blob/master/docs/rules/naming/named-parameters-mapping.md) rule. A shout-out to @0xteddybear for noticing it.

**Review by 0xteddybear — DISMISSED** [review]
:rocket:

File: src/interfaces/.solhint.json (hunk 1 above)
Comment (0xJabberwock) [reply src/interfaces/.solhint.json:3]: I like having a standard for interface ordering; following the layout from the Solidity docs convinces me—at the same time, I prefer putting default getters above other functional logic. I think this last preference conflicts with how `ordering` works.

File: .solhint.json [hunk 3]
```diff
@@ -1,18 +1,14 @@
...
-    "compiler-version": ["off"],
-    "constructor-syntax": "warn",
-    "quotes": ["error", "single"],
+    "compiler-version": ["warn"],
+    "quotes": "off",
     "func-visibility": ["warn", { "ignoreConstructors": true }],
-    "not-rely-on-time": "off",
```
Comment (gas1cent) [inline .solhint.json:8]: @0xteddybear why removing this rule ser?

Comment (0xteddybear) [reply .solhint.json:8]: it's not necessary to explicitly disable it since it was removed from solhint:recommended

**Review by gas1cent — APPROVED** [review]

**Review by 0xJabberwock — APPROVED** [review]

File: test/.solhint.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "rules": {
+    "style-guide-casing": [
+      "warn",
+      {
+        "ignorePublicFunctions":true,
+        "ignoreExternalFunctions":true,
+        "ignoreContracts":true
+      }
+    ],
+    "no-global-import": "off",
```
[the remark this answers is not in the document]
Comment (0xJabberwock) [reply test/.solhint.json:11]: I generally support disabling it for tests or scripts, as it is handy to import `forge-std/Test.sol` or `forge-std/Script.sol` entirely.

**Pull request merged**